lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -565,6 +566,9 @@ function dosomething_global_redirect_node_edit($node, $user_language) {
     $new_url = str_replace(' ', '', '/node/' . $node->nid . '/edit/' . $lang_append);
     drupal_goto($new_url);
   }
+  else if (dosomething_global_get_current_prefix() == '' && $prefix != '') {
```

Comment:
Then better use `===` operand and explain the logic in comment.
Also, `!empty($prefix)`.